sources/api/schnauzer/src/bin/schnauzer-v2/clirequirements.rs

```diff
@@ -0,0 +1,172 @@
+        let re_captures =
+            EXTENSION_REQUIREMENT
+                .captures(s)
+                .context(error::RequirementsParseSnafu {
+                    requirement: s.to_string(),
+                    reason: "Extension requirement regex did not find a match",
+                })?;
+
+        let name = re_captures
+            .name("extension")
+            .context(error::RequirementsParseSnafu {
+                requirement: s.to_string(),
+                reason: "Did not find extension name",
+            })?
+            .as_str()
+            .to_string();
+
+        let version = re_captures
+            .name("version")
+            .context(error::RequirementsParseSnafu {
+                requirement: s.to_string(),
+                reason: "Did not find extension version",
+            })?
+            .as_str()
+            .to_string();
+
+        let helpers = re_captures
+            .name("helpers")
+            .map(|helpers| {
+                split_helpers(helpers.as_str())
+                    .into_iter()
+                    .filter(|helper| !helper.is_empty())
+                    .map(str::to_string)
+                    .collect()
+            })
+            .unwrap_or_default();
+
+        Ok(CLIExtensionRequirement(ExtensionRequirement {
+            name,
+            version,
+            helpers,
+        }))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use schnauzer::v2::ExtensionRequirement;
+
+    use super::CLIExtensionRequirement;
+
+    #[test]
+    fn test_requirements_parse_succeeds() {
+        let test_cases = &[
+            (
+                "extension@version(helper1)",
+                ExtensionRequirement {
+                    name: "extension".to_string(),
+                    version: "version".to_string(),
+                    helpers: vec!["helper1".to_string()],
+                },
+            ),
+            (
+                " myextension@v1(    )   ",
+                ExtensionRequirement {
+                    name: "myextension".to_string(),
+                    version: "v1".to_string(),
+                    helpers: vec![],
+                },
+            ),
+            (
+                "extension@version",
+                ExtensionRequirement {
+                    name: "extension".to_string(),
+                    version: "version".to_string(),
+                    helpers: vec![],
+                },
+            ),
+            (
+                "std@v1   (   base64_decode,    join_array  )  ",
+                ExtensionRequirement {
+                    name: "std".to_string(),
+                    version: "v1".to_string(),
+                    helpers: vec!["base64_decode", "join_array"]
+                        .into_iter()
+                        .map(String::from)
+                        .collect(),
+                },
+            ),
+            (
+                "weird-extension@but_valid1.23(1, 2,   3)",
```

Comment:
I don't feel strongly about this, but for CLI input I tend to avoid anything that might get mangled or misinterpreted if pasted into a prompt or script. Parens and whitespace jump out at me as slightly problematic.

Would something like `weird-extension@but_valid1.23+1,2,3` ("add the 1, 2, and 3 helpers from this version of this extension") support the requirements?

Reply:
It should, though I'm conflicted. I've been thinking about this syntax since I submitted this PR.

The `requires` line is meant to basically replicate the functionality of frontmatter. One thing it doesn't do here is allow the `optional = true` field, which is probably fine; however, I don't want to be trapped in a corner of never being able to support it.

I've been thinking we may need something like `extension@version(helpers=[helper1, helper2, ...])` so that we can add other named parameters if needed later. Of course, if the time comes, a regex would make for a terrible parser, but that's a bridge for another time.

All this to say, we could probably still use different separators to avoid parens and whitespace, maybe something like:

```
extension@version+helpers:1,2,3+optional:true
```

What do you think?

Reply:
Hmm, on the other hand, it's extremely unlikely that the templates themselves will be CLI-friendly, they almost always contain whitespace and curly brace characters.

I'm erring on continuing to use the parens (I personally find it more readable) but adding the named arguments.